MDANSE/Src/MDANSE/Mathematics/Signal.py

```diff
@@ -308,3 +318,932 @@ def get_spectrum(signal, window=None, timeStep=1.0, axis=0, fft="fft"):
+    def polynomial_string(coeffs, unit, analog: bool = True) -> str:
+        """Formats a polynomial into a string that has a symbolic mathematical appearance.
+
+        :Parameters:
+            #. coeffs (np.ndarray): Array of polynomial coefficients
+            #. unit (str): String representation of the mathematical symbol corresponding to the S or Z planes
+        :Returns:
+            #. str: Symbolic polynomial string
+        """
+        if not coeffs.any():
+            return ""
+        order = len(coeffs) - 1
+        expr = ""
+
+        for idx, coeff in enumerate(coeffs):
+            power = order - idx if analog else -idx
+            if coeff != 0:
+                if idx > 0 and coeff > 0:
+                    expr += " + "
+                elif idx > 0 and coeff < 0:
+                    expr += " - "
+
+                abs_coeff = abs(coeff)
+                if power == 0:
+                    expr += f"{abs_coeff:.3f}"
+                elif power == 1:
+                    expr += f"{abs_coeff:.3f}*({unit})"
+                else:
+                    expr += f"{abs_coeff:.3f}*({unit})^{power}"
+        return expr
+
+    @classmethod
+    def rational_polynomial_string(
+        cls, numerator, denominator, analog=True
+    ) -> dict[str, str]:
+        """Formats a transfer function rational polynomial into a pair of strings.
+
+        :Parameters:
+            #. numerator (np.ndarray): Array of coefficients representing the numerator of the transfer function
+            #. denominator (np.ndarray): Array of coefficients representing the denominator of the transfer function
+            #. analog (bool): The filter is analog (Laplace/S-domain) or digital (Z-domain)
+        :Returns:
+            #. dict[str, str]: Dictionary of string coefficients representing the transfer function rational polynomial
+        """
+        if analog:
+            # Analogue (Laplace-domain) transfer function
+            numerator_str = cls.polynomial_string(numerator, cls.S)
+            denominator_str = Filter.polynomial_string(denominator, cls.S)
+            return {
+                "unit": "S",
+                "numerator": numerator_str,
+                "denominator": denominator_str,
+            }
+
+        # Digital (Z-domain) transfer function
+        numerator_str = Filter.polynomial_string(numerator, cls.Z, False)
+        denominator_str = Filter.polynomial_string(denominator, cls.Z, False)
+        return {"unit": "Z", "numerator": numerator_str, "denominator": denominator_str}
+
+    def attributes_to_string(self, description) -> str:
+        """Formats the given filter attribute into a description string.
+
+        :Parameters:
+            #. description (str): The description of the attribute as a multiline string
+        :Returns:
+            #. str: The description string concatenated with substrings for each filer attribute
+        """
+        settings = self.__class__.__dict__["default_settings"]
```

Comment:
Although `self` has access to all the class attributes, so probably not necessary.
```suggestion
        settings = type(self).__dict__["default_settings"]
```